# -*- coding: utf-8 -*-
"""homeworkDB.ipynb

Automatically generated by Colaboratory.

Original file is located at
    https://colab.research.google.com/drive/1VDoAVu_I2re5KFWy8gXQV4raMIXXgoA_
"""

# Создание БД
import sqlite3

connection = sqlite3.connect('Students.db')
cursor = connection.cursor()
query = """CREATE TABLE Students (
Student_Id INTEGER NOT NULL PRIMARY KEY,
Student_Name TEXT NOT NULL,
School_Id INTEGER NOT NULL
);
"""
cursor.execute(query)
connection.commit()
connection.close()

#Заполнение БД
import sqlite3

connection = sqlite3.connect('Students.db')
cursor = connection.cursor()
query = """INSERT INTO Students (Student_Id, Student_Name, School_Id)
VALUES
(201, 'Иван', 1),
(202, 'Петр', 2),
(203, 'Анастасия', 3),
(204, 'Игорь', 4)
"""
cursor.execute(query)
connection.commit()
connection.close()

#Работа с БД

import sqlite3

def get_connection():
  connection = sqlite3.connect('Students.db')
  return connection

def close_connection(connection):
  if connection:
    connection.close()

def get_Student_detail(Student_id):
  try:
    connection = get_connection()
    cursor = connection.cursor()
    select_query = """SELECT * FROM Students WHERE Student_Id = ?"""
    cursor.execute(select_query, (Student_id,))
    records = cursor.fetchall()
    print ("Данные о студенте")
    for row in records:
      print ("ID студента", row[0])
      print ("Имя студента", row[1])
      print ("ID школы", row[2])
    close_connection(connection)
  except (Exception, sqlite3.Error) as error:
    print ("Ошибка в получении данных по школе ", error)


get_Student_detail(204)